src/test/java/com/palantir/ri/ResourceIdentifierTest.java

```diff
@@ -192,70 +187,101 @@ void testSerialization() throws IOException {
-        ResourceIdentifier value2 = om.readValue(serializedString2, ResourceIdentifier.class);
-        ResourceIdentifier value3 = om.readValue(serializedString3, ResourceIdentifier.class);
-        assertEquals(toJsonString(ridString), serializedString);
-        assertEquals(toJsonString(ridString1), serializedString1);
-        assertEquals(toJsonString(ridString2), serializedString2);
-        assertEquals(toJsonString(ridString3), serializedString3);
-        assertEquals(rid, value);
-        assertEquals(rid1, value1);
-        assertEquals(rid2, value2);
-        assertEquals(rid3, value3);
+        assertThat(om.writeValueAsString(rid)).isEqualTo(toJsonString(ridString));
+        assertThat(om.writeValueAsString(rid1)).isEqualTo(toJsonString(ridString1));
+        assertThat(om.writeValueAsString(rid2)).isEqualTo(toJsonString(ridString2));
+        assertThat(om.writeValueAsString(rid3)).isEqualTo(toJsonString(ridString3));
+        assertThat(om.readValue(om.writeValueAsString(rid), ResourceIdentifier.class))
+                .isEqualTo(rid);
+        assertThat(om.readValue(om.writeValueAsString(rid1), ResourceIdentifier.class))
+                .isEqualTo(rid1);
+        assertThat(om.readValue(om.writeValueAsString(rid2), ResourceIdentifier.class))
+                .isEqualTo(rid2);
+        assertThat(om.readValue(om.writeValueAsString(rid3), ResourceIdentifier.class))
+                .isEqualTo(rid3);
     }
 
     private String toJsonString(String string) {
-        return String.format("\"%s\"", string);
+        return "\"" + string + "\"";
     }
 
     @Test
     void testStringConstruction() {
-        assertEquals(
-                "ri.service..type.name",
-                ResourceIdentifier.of("service", "", "type", "name").toString());
-        assertEquals(
-                "ri.service.instance.type.name",
-                ResourceIdentifier.of("service", "instance", "type", "name").toString());
+        assertThat(ResourceIdentifier.of("service", "", "type", "name").toString())
+                .isEqualTo("ri.service..type.name");
+        assertThat(ResourceIdentifier.of("service", "instance", "type", "name").toString())
+                .isEqualTo("ri.service.instance.type.name");
     }
 
     @Test
     void testStringConstructionWithMultipleLocatorComponents() {
-        assertEquals(
-                "ri.service..type.name1",
-                ResourceIdentifier.of("service", "", "type", "name1").toString());
-        assertEquals(
-                "ri.service..type.name1.name2",
-                ResourceIdentifier.of("service", "", "type", "name1", "name2").toString());
-        assertEquals(
-                "ri.service..type.name1.name2.name3",
-                ResourceIdentifier.of("service", "", "type", "name1", "name2", "name3")
-                        .toString());
+        assertThat(ResourceIdentifier.of("service", "", "type", "name1").toString())
```

Comment:
you could just do `.hasToString`, you use the inverse in `assertNotEqualsAndDifferentHashCode` below